pkg/plugins/resources/cargopackage/condition.go

```diff
@@ -0,0 +1,66 @@
+package cargopackage
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/pipeline/scm"
+	"github.com/updatecli/updatecli/pkg/core/result"
+)
+
+// Condition checks that a git tag exists
+func (cp *CargoPackage) Condition(source string) (bool, error) {
+	if len(cp.spec.IndexDir) == 0 {
+		// No IndexDir provided nor scm resource
+		// We should use the default `crates.io` repository
+		indexDir, err := loadDefaultCrateIndex()
+		if err != nil {
+			return false, err
+		}
+		cp.spec.IndexDir = indexDir
+	}
+	return cp.condition(source)
+}
+
+// ConditionFromSCM test if a tag exists from a git repository specific from SCM
+func (cp *CargoPackage) ConditionFromSCM(source string, scm scm.ScmHandler) (bool, error) {
+	path := scm.GetDirectory()
+
+	if len(cp.spec.IndexDir) > 0 {
+		logrus.Warningf("IndexDir is defined and set to %q but is overridden by the scm definition %q",
+			cp.spec.IndexDir,
+			path)
+	}
+	cp.spec.IndexDir = path
```

Comment:
We try to avoid modifying the spec parameter.
The reason is at any moment in time we want to be able to identify the user input. It's more a convention in the code base 

Instead we copy the value to a new variable and then that variable can be change. If needed we still have the initial value in the spec

Reply:
In this case that would look like `cp.IndexDir = path` where `cp.IndexDir` is init in the New function